espi-service/src/espi_service.rs

```diff
@@ -156,64 +160,64 @@ impl Service<'_, '_> {
-            let pkt_start_index = 64 * i;
-            let pkt_end_index = min(response_len, pkt_start_index + 64);
-            let mctp_pkt_len = pkt_end_index - pkt_start_index;
-            trace!(
-                "start: {}, end: {}, len: {}",
-                pkt_start_index, pkt_end_index, mctp_pkt_len
-            );
-            let som = i == 0;
-            let eom = i == num_pkts - 1;
-
-            if let Ok((final_packet, final_packet_size)) =
-                // take into account response_len
-                build_mctp_header(
-                    &acpi_pkt[pkt_start_index..pkt_end_index],
-                    mctp_pkt_len,
-                    endpoint,
-                    som,
-                    eom,
-                )
-            {
-                info!("Sending MCTP response: {:?}", &final_packet[..final_packet_size]);
-
-                // SAFETY: Safe as the access to espi is protected by a mut reference.
-                let result = unsafe { espi.oob_get_write_buffer(OOB_PORT_ID) };
+        let mut mctp_ctx =
+            mctp_rs::MctpPacketContext::new(mctp_rs::medium::smbus_espi::SmbusEspiMedium, &mut pkt_ctx_buf);
+
+        let reply_context: mctp_rs::MctpReplyContext<SmbusEspiMedium> = mctp_rs::MctpReplyContext {
+            source_endpoint_id: mctp_rs::endpoint_id::EndpointId::Id(0x80),
+            destination_endpoint_id: match endpoint {
+                EndpointID::Internal(Internal::Battery) => mctp_rs::endpoint_id::EndpointId::Id(8),
+                EndpointID::Internal(Internal::Thermal) => mctp_rs::endpoint_id::EndpointId::Id(9),
+                EndpointID::Internal(Internal::Debug) => mctp_rs::endpoint_id::EndpointId::Id(10),
+                _ => mctp_rs::endpoint_id::EndpointId::Id(0x80),
+            },
+            packet_sequence_number: mctp_rs::mctp_sequence_number::MctpSequenceNumber::new(0),
+            message_tag: mctp_rs::mctp_message_tag::MctpMessageTag::try_from(3).unwrap(),
+            medium_context: SmbusEspiReplyContext {
+                destination_slave_address: 1,
+                source_slave_address: 0,
+            }, // Medium-specific context
+        };
 
-                match result {
-                    Ok(dest_slice) => {
-                        dest_slice[..final_packet_size].copy_from_slice(&final_packet[..final_packet_size]);
-                    }
-                    Err(_e) => {
-                        #[cfg(feature = "defmt")]
-                        error!("Failed to retrieve OOB write buffer: {}", _e);
-                        // TODO: Ask if we need to send a response if the request is malformed
-                        return;
-                    }
+        let mut packet_state = mctp_ctx
+            .serialize_packet(reply_context, &acpi_pkt[..response_len])
+            .unwrap();
+        // Send each packet
+        while let Some(packet_result) = packet_state.next() {
+            let packet = packet_result.unwrap();
```

Comment:
If `packet_result` is a `None`, this will `panic!()`. Probably a good idea to avoid `panic!()`s